essentials/src/types.rs

```diff
@@ -31,3 +32,35 @@ pub type AccountId32 = utils::AccountId32;
 pub type Timestamp = u64;
 pub type SessionKeys = runtime::SessionKeys;
 pub type SubxtCall = runtime::RuntimeCall;
+
+pub type ClaimQueue = BTreeMap<u32, VecDeque<Option<ParasEntry>>>;
+
+// TODO: Take it from runtime types v5
+/// Polkadot v5 ParasEntry type
+#[derive(Debug)]
+pub struct ParasEntry {
+	/// The `Assignment`
+	pub assignment: Assignment,
+	/// Number of times this has been retried.
+	pub retries: u32,
+	/// The block height where this entry becomes invalid.
+	pub ttl: BlockNumber,
+}
+
+// TODO: Take it from runtime types v5
+/// Polkadot v5 Assignment type
+#[derive(Debug)]
+pub struct Assignment {
+	/// Assignment's ParaId
+	pub para_id: u32,
+}
+
+// TODO: Take it from runtime types v5
+/// Temporary abstraction to cover core state until v5 types are released
+#[derive(Debug)]
+pub enum CoreOccupied {
+	/// The core is not occupied.
+	Free,
+	/// A paras.
```

Comment:
The naming is a bit confusing, Is it occupied by a parachain or many ?

Reply:
It's an original name from Polkadot (on-demand pallet branch)

Reply:
It's an original name from Polkadot (on-demand pallet branch)